pkg/shardddl/pessimism/lock.go

```diff
@@ -100,6 +100,19 @@ func (l *Lock) ForceSynced() {
 	l.remain = 0
 }
 
+// RevertSynced reverts the synced stage of the sources.
+func (l *Lock) RevertSynced(sources []string) {
+	l.mu.Lock()
+	defer l.mu.Unlock()
+
+	for _, source := range sources {
+		if synced, ok := l.ready[source]; ok && synced {
+			l.ready[source] = false
```

Comment:
do we need to revert `l.done[source]`

Reply:
The only usage of `RevertSynced` is to revert `Synced` for `ForceSynced`.
and currenly we can't `RevertSynced` is the owner done the exec operation (`done` happened after that).